client/src/redux/actions/observations.js

```diff
@@ -1,57 +1,33 @@
-export const SAVING = 'SAVING';
-export const IDLE = 'IDLE';
-export const LOADING = 'LOADING';
-export const OBSERVATIONS_LOADED = 'OBSERVATIONS_LOADED';
-export const NUM_OBSERVATIONS = 'NUM_OBSERVATIONS';
-export const SET_OBSERVATIONS = 'SET_OBSERVATIONS';
-
-export function setNumObservations(numObservations) {
-  return {
-    type: NUM_OBSERVATIONS,
-    numObservations,
-  };
-}
-
-export function setObservations(observations) {
-  return {
-    type: SET_OBSERVATIONS,
-    observations,
-  };
-}
-
-export const addObservation = (observation, observationNumber) => (dispatch, getState) => {
-  const { observations } = getState().observationsReducer;
-  dispatch(setObservations(observations.concat(observation)));
-  dispatch(setNumObservations(observationNumber));
-};
-
-export const fetchObservations = userSession => async dispatch => {
-  const fileNames = [];
-  const observations = [];
-  let numObservations = 0;
-
-  // Get the observation file names
-  await userSession.listFiles(fileName => {
-    if (fileName.includes('observation/')) {
-      fileNames.push(fileName);
+import { getObservations, postObservation } from '../../utils/blockstackHelpers';
+import { clearSurvey } from './survey';
+
+export const ADD_OBSERVATION = 'ADD_OBSERVATION';
+export const FETCH_OBSERVATIONS = 'FETCH_OBSERVATIONS';
+
+export const addObservation = (userSession, observation) => async dispatch => {
+  const obs = await getObservations(userSession);
+  let obsArray;
+  let fileNumber = 1;
+  if (obs) {
+    const currentArray = JSON.parse(obs);
+    fileNumber = currentArray.length + 1;
+    obsArray = [...currentArray, observation];
+  } else {
+    obsArray = [observation];
+  }
+  postObservation(userSession, obsArray, fileNumber).then(didPost => {
+    if (didPost) {
+      dispatch({ type: ADD_OBSERVATION, payload: observation });
     }
-    return true;
   });
 
-  // Prepare to get each observation file and determine the number of observations
-  const getFilePromises = fileNames.map(fileName => {
-    const observationNum = parseInt(fileName.replace(/^\D+/g, ''), 10);
-    numObservations = observationNum;
-    return userSession.getFile(fileName);
-  });
-
-  dispatch(setNumObservations(numObservations));
-
-  // Convert the observations to JSON objects
-  (await Promise.all(getFilePromises)).forEach(observationString => {
-    const observation = JSON.parse(observationString);
-    observations.push(observation);
-  });
+  dispatch(clearSurvey());
```

Comment:
_I feel the addObservation thunk should have nothing to do with a survey_

I'm not sure that I totally understand... an observation is the survey data.  The addObservation thunk is the submission of a survey.  So clear survey (not a function I wrote, I'm just using the actions that are already built to manage the survey ui state) would follow the submission of the survey.  

I can certainly see the argument for clarity sake of moving the clearSurvey outside the submit action and just call it directly after in the component.

Reply:
Yeah, I guess it's just the naming that's throwing me off. I see that the only way to add an observation is to submit a survey, but when I think of a function `addObservation`, I wouldn't think it touches the survey, like with `clearSurvey`.

> I can certainly see the argument for clarity sake of moving the clearSurvey outside the submit action and just call it directly after in the component.

This is really what I'm getting at.

Reply:
I agree with the above ^ good pointing out. If you're willing I think it would be worthwhile to move `clearSurvey` outside of this action, although we can make another issue for it as well.

Reply:
I'll move it outside the action, back to the component.